cpp/benchmarks/ast/polynomials.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmarks/common/generate_input.hpp>
+
+#include <cudf/ast/expressions.hpp>
+#include <cudf/column/column.hpp>
+#include <cudf/table/table.hpp>
+#include <cudf/transform.hpp>
+#include <cudf/utilities/error.hpp>
+
+#include <thrust/iterator/counting_iterator.h>
+
+#include <nvbench/nvbench.cuh>
+#include <nvbench/types.cuh>
+
+template <typename key_type>
+static void BM_ast_polynomials(nvbench::state& state)
+{
+  auto const num_rows = static_cast<cudf::size_type>(state.get_int64("num_rows"));
+  auto const order    = static_cast<cudf::size_type>(state.get_int64("order"));
+
+  CUDF_EXPECTS(order > 0, "Polynomial order must be greater than 0");
+
+  data_profile profile;
+  profile.set_distribution_params(cudf::type_to_id<key_type>(),
+                                  distribution_id::NORMAL,
+                                  static_cast<key_type>(0),
+                                  static_cast<key_type>(1));
+  auto table = create_random_table({cudf::type_to_id<key_type>()}, row_count{num_rows}, profile);
+  auto column_view = table->get_column(0);
+
+  std::vector<cudf::numeric_scalar<key_type>> constants;
+
+  std::transform(thrust::make_counting_iterator(0),
+                 thrust::make_counting_iterator(order + 1),
+                 std::back_inserter(constants),
+                 [](int) { return cudf::numeric_scalar<key_type>(1); });
```

Comment:
```suggestion
  std::vector<cudf::numeric_scalar<key_type>> constants(order + 1);

  std::fill(constants.begin(),
                 constants.end(),
                 cudf::numeric_scalar<key_type>(1));
```

Reply:
it's temporary in the mean time as we intend to use a random constant generator to fill the range

Reply:
Ah I see, makes sense. Thanks for confirming!

Reply:
will do

Reply:
done